tests/samplers_tests/tpe_tests/test_parzen_estimator.py

```diff
@@ -239,6 +261,29 @@ def test_calculate_shape_check(
     assert len(s_sigmas) == len(mus) + int(prior) if len(mus) > 0 else len(mus) + 1
 
 
+@pytest.mark.parametrize("prior_weight", [None, -1.0, 0.0])
+@pytest.mark.parametrize("mus", (np.asarray([]), np.asarray([0.4]), np.asarray([-0.4, 0.4])))
+def test_invalid_prior_weight(prior_weight: float, mus: np.ndarray) -> None:
```

Comment:
[memo] If we add an invalid `prior_weight`, we should raise an error. To be fixed.

Reply:
[FYI] Maybe we should allow `prior_weight == 0.0`, since this is arguably one natural choice.

Reply:
Should we add `-inf` or `nan` in `prior_weight` as well? or possibly `+inf` also?